modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/utils/iframes.js

```diff
@@ -14,43 +14,39 @@
 
 import {OPEN_MODAL, OPEN_MODAL_FROM_IFRAME} from './eventsDefinitions';
 
-const iframeHandlerModalNamespace = 'iframe-handler-modal_';
-let modalsCounter = 0;
-const registeredModals = new Map();
+export function isPageInIframe() {
+	return window.location !== window.parent.location;
+}
 
-Liferay.on(OPEN_MODAL_FROM_IFRAME, (payload) => {
-	let firstAvailableModalId = null;
+const registeredModals = new Set();
+let modalsCounter = 0;
 
-	// eslint-disable-next-line no-for-of-loops/no-for-of-loops
-	for (const [modalId, modalRef] of registeredModals.entries()) {
-		if (modalRef) {
-			firstAvailableModalId = modalId;
-			break;
+if(!isPageInIframe()) {
+	Liferay.on(OPEN_MODAL_FROM_IFRAME, (payload) => {
+		if(!registeredModals.size) {
+			throw new Error(`No registered modals founds`)
 		}
-	}
-
-	window.top.Liferay.fire(OPEN_MODAL, {
-		...payload,
-		id: firstAvailableModalId,
+	
+		const modalsArray = Array.from(registeredModals);
+	
+		Liferay.fire(OPEN_MODAL, {
+			...payload,
+			id: modalsArray[modalsArray.length - 1],
+		});
 	});
-});
+}
 
-Liferay.on('endNavigate', () => {
+Liferay.on('beforeNavigate', (payload) => {
 	registeredModals.clear();
 });
```

Comment:
This code (L24-L41) seems wrong in this file. It looks like it belongs in a life-cycle of a component. 

It's kinda ok that this file contains a closed global store (`registeredModals`), with API to access it (`subscribeModal`, `unsubscribeModal`, `getRegisteredModals`). I'm not sure if that's the ideal solution :thinking:

Reply:
Mhhh maybe the ideal solution would be getting rid of the entire file and using the global store. It's a bit out of the scope of this PR so I would keep it this way for now (but I'll definitely have a look at it).

Reply:
@FabioDiegoMastrorilli Sure, refactoring this to global store does make sense. Also, you are right that it is beyond the scope of this PR. We can leave `registeredModals` in this file, for now.

But, event listeners definition (L24-L41) does not belong in this file. This is an eagerly executed code in a utility file that serves as global scope. This code should probably be in constructor of `SidePanel.js`. To access `registeredModals`, we can export a getter, `getRegisteredModals`.